# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/vm/SharedArrayObject.cpp
# Commit: 75d829afb4b0
# Full Hash: 75d829afb4b06a3a8b7e91ea15c8a0cf34665fb7
# Author: Cosmin Sabou <csabou@mozilla.com>
# Date: 2025-09-16 12:02 +0300
# Description:
#   Revert "Bug 1969509 - wasm: Update .maxByteLength to report the source wasm max pages in bytes. r=rhunt" for causing javascript engine crashes (bug 1988694)
# ==============================================================================

diff -r 896c8ee9171d -r 75d829afb4b0 js/src/vm/SharedArrayObject.cpp
--- a/js/src/vm/SharedArrayObject.cpp	Tue Sep 16 04:43:56 2025 +0300
+++ b/js/src/vm/SharedArrayObject.cpp	Tue Sep 16 12:02:13 2025 +0300
@@ -331,28 +331,6 @@
   MOZ_ASSERT(IsSharedArrayBuffer(args.thisv()));
   auto* buffer = &args.thisv().toObject().as<SharedArrayBufferObject>();
 
-  // Special case for wasm with potentially 64-bits memory.
-  // Manually compute maxByteLength to avoid an overflow on 32-bit machines.
-  if (buffer->isWasm()) {
-    auto* wasmBuffer = buffer->rawWasmBufferObject();
-
-    args.rval().setNumber(
-        double(wasmBuffer->wasmSourceMaxPages().value() * wasm::PageSize));
-    return true;
-  }
-  // Special case for wasm with potentially 64-bits memory.
-  // Manually compute the maxByteLength to avoid an overflow on 32-bit machines.
-  if (buffer->isWasm()) {
-    Pages sourceMaxPages = buffer->rawWasmBufferObject()->wasmSourceMaxPages();
-    uint64_t sourceMaxBytes = sourceMaxPages.byteLength64();
-
-    MOZ_ASSERT(sourceMaxBytes <=
-               wasm::PageSize * wasm::MaxMemory64PagesValidation);
-    args.rval().setNumber(double(sourceMaxBytes));
-
-    return true;
-  }
-
   // Steps 4-6.
   args.rval().setNumber(buffer->byteLengthOrMaxByteLength());
   return true;